test/quantization/test_quantize_fx.py

```diff
@@ -757,3 +768,177 @@ def forward(self, x):
+            original_out = graph_module(input_value)
+            is_not_tuple_out = not isinstance(original_out, tuple)
+            script_out = script(input_value)
+            self.assertEqual(
+                (original_out - script_out).abs().max(), 0,
+                'Reslut of original graph module and script module does not match')
+
+            # set to train just before quantization
+            if mode != 'static':
+                model.train()
+
+            graph_module = fuse(graph_module)
+            quantizer = Quantizer()
+            prepared = quantizer.prepare(graph_module, qconfig_dict)
+
+            if mode == 'ddp':
+                mp.spawn(run_ddp,
+                         args=(world_size, prepared),
+                         nprocs=world_size,
+                         join=True)
+            elif mode == 'qat':
+                assert prepared.training, 'prepared must be in training mode for qat'
+                optimizer = torch.optim.SGD(prepared.parameters(), lr=0.0001)
+                criterion = nn.CrossEntropyLoss()
+                train_one_epoch(prepared, criterion, optimizer, [(input_value, output_value)], torch.device('cpu'), 1)
+            else:
+                for i in range(10):
+                    prepared(input_value)
+
+            # print('after observation root:', prepared.root)
+
+            qgraph = quantizer.convert(prepared)
+            # print('after quantization root:', qgraph.root)
+            # print('after quantization code:', qgraph.src)
+            qgraph.eval()
+            qgraph_script = torch.jit.script(qgraph)
+            # print('quantized and scripted:', qgraph_script.graph)
+
+            qgraph_out = qgraph(input_value)
+            qgraph_script = qgraph_script(input_value)
+
+            if is_not_tuple_out:
+                diff_of_quant[mode][name] = (original_out - qgraph_out).abs().max()
+                if diff_of_quant[mode][name] > 1:
+                    greater_than_one[mode][name] = diff_of_quant[mode][name]
+                    assert torch.allclose(qgraph_out, qgraph_script), 'graph, scripted graph'
+            else:
+                print('tuple output')
+
+            if name in quantized_model_list:
+                if (mode, name) in quantized_not_working:
+                    continue
+                # comparing to eager mode quantization
+                qeager = quantized_models.__dict__[name](pretrained=True, quantize=False).eval().float()
+                ref_out = qeager(input_value)
+                qeager.qconfig = qconfig
+                if mode == 'static':
+                    qeager.fuse_model()
+                    prepare(qeager, inplace=True)
+                else:
+                    qeager.train()
+                    qeager.fuse_model()
+                    prepare_qat(qeager, inplace=True)
+
+                # calibration
+                if mode == 'ddp':
+                    mp.spawn(run_ddp,
```

Comment:
Will this run on all combinations of CPU/GPU and OS?

Reply:
no, will only test on multi gpu setting, I will split this to a separate test in the update